src/quad-tester/tester2simdqp.c

```diff
@@ -0,0 +1,326 @@
+#endif
+
+#ifdef ENABLE_PURECFMA_SCALAR
+#define CONFIG 2
+#include "helperpurec_scalar.h"
+#include "qrenamepurecfma_scalar.h"
+#endif
+
+#ifdef ENABLE_SSE2
+#define CONFIG 2
+#include "helpersse2.h"
+#include "qrenamesse2.h"
+#endif
+
+#ifdef ENABLE_AVX2128
+#define CONFIG 1
+#include "helperavx2_128.h"
+#include "qrenameavx2128.h"
+#endif
+
+#ifdef ENABLE_AVX
+#define CONFIG 1
+#include "helperavx.h"
+#include "qrenameavx.h"
+#endif
+
+#ifdef ENABLE_FMA4
+#define CONFIG 4
+#include "helperavx.h"
+#include "qrenamefma4.h"
+#endif
+
+#ifdef ENABLE_AVX2
+#define CONFIG 1
+#include "helperavx2.h"
+#include "qrenameavx2.h"
+#endif
+
+#ifdef ENABLE_AVX512F
+#define CONFIG 1
+#include "helperavx512f.h"
+#include "qrenameavx512f.h"
+#endif
+
+#ifdef ENABLE_ADVSIMD
+#define CONFIG 1
+#include "helperadvsimd.h"
+#include "qrenameadvsimd.h"
+#endif
+
+#ifdef ENABLE_SVE
+#define CONFIG 1
+#include "helpersve.h"
+#include "qrenamesve.h"
+#endif
+
+#ifdef ENABLE_VSX
+#define CONFIG 1
+#include "helperpower_128.h"
+#include "qrenamevsx.h"
+#endif
+
+#ifdef ENABLE_DSP128
+#define CONFIG 2
+#include "helpersse2.h"
+#include "qrenamedsp128.h"
+#endif
+
+#ifdef ENABLE_DSP256
+#define CONFIG 1
+#include "helperavx.h"
+#include "qrenamedsp256.h"
+#endif
+
+//
+
+#define DENORMAL_DBL_MIN (4.9406564584124654418e-324)
+
+#define POSITIVE_INFINITY INFINITY
+#define NEGATIVE_INFINITY (-INFINITY)
+
+typedef union {
+  Sleef_quad q;
+  xuint128 x;
+  struct {
+    uint64_t l, h;
+  };
+} cnv_t;
+
+Sleef_quad nexttoward0q(Sleef_quad x, int n) {
+  cnv_t cx;
+  cx.q = x;
+  cx.x = add128(cx.x, xu(n < 0 ? 0 : -1, -(int64_t)n));
+  return cx.q;
+}
+
+static vargquad vset(vargquad v, int idx, Sleef_quad d) { v.s[idx] = d; return v; }
+static Sleef_quad vget(vargquad v, int idx) { return v.s[idx]; }
+
+static int vgeti(vint v, int idx) {
+  int a[VECTLENDP*2];
+  vstoreu_v_p_vi(a, v);
+  return a[idx];
+}
+
+int main(int argc,char **argv)
+{
+  mpfr_set_default_prec(1024);
+  xsrand(time(NULL) + (((int)getpid()) << 12));
+  srandom(time(NULL) + (((int)getpid()) << 12));
+
+  //
+
+  const Sleef_quad oneEMinus10Q  = cast_q_str("1e-10");
+  const Sleef_quad oneEPlus10Q   = cast_q_str("1e+10");
+  const Sleef_quad oneEMinus100Q = cast_q_str("1e-100");
+  const Sleef_quad oneEPlus100Q  = cast_q_str("1e+100");
+  const Sleef_quad oneEMinus1000Q = cast_q_str("1e-1000");
+  const Sleef_quad oneEPlus1000Q  = cast_q_str("1e+1000");
+  const Sleef_quad quadMin = cast_q_str("3.36210314311209350626267781732175260e-4932");
+  const Sleef_quad quadMax = cast_q_str("1.18973149535723176508575932662800702e+4932");
+  const Sleef_quad quadDenormMin = cast_q_str("6.475175119438025110924438958227646552e-4966");
+
+  //
+
+  int cnt, ecnt = 0;
+  vargquad a0, a1, a2, a3;
+  Sleef_quad q0, q1, q2, q3, t;
+  mpfr_t frw, frx, fry, frz;
+  mpfr_inits(frw, frx, fry, frz, NULL);
+
+  for(cnt = 0;ecnt < 1000;cnt++) {
+    int e = cnt % VECTLENDP;
+
+    switch(cnt & 127) {
+    case 127:
```

Comment:
The `case` values should be handle by enums, with description of the meaning, otherwise it will be difficult to know what is going on here.